core-processor/src/processor.rs

```diff
@@ -17,48 +17,50 @@
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 use crate::{
-    common::{
-        Dispatch, DispatchKind, DispatchOutcome, DispatchResult, DispatchResultKind, JournalNote,
-    },
+    common::{DispatchOutcome, DispatchResult, DispatchResultKind, JournalNote},
     configs::{BlockInfo, ExecutionSettings},
     executor,
     ext::Ext,
 };
 use alloc::{collections::BTreeMap, vec::Vec};
 use gear_backend_common::Environment;
 use gear_core::{
-    message::Message,
+    message::{Dispatch, DispatchKind, Message},
     program::{Program, ProgramId},
 };
 
 /// Process program & dispatch for it and return journal for updates.
 pub fn process<E: Environment<Ext>>(
-    program: Program,
+    program: Option<Program>,
     dispatch: Dispatch,
     block_info: BlockInfo,
 ) -> Vec<JournalNote> {
-    let execution_settings = ExecutionSettings::new(block_info);
-    let initial_nonce = program.message_nonce();
-
-    match executor::execute_wasm::<E>(program, dispatch.clone(), execution_settings) {
-        Ok(res) => match res.kind {
-            DispatchResultKind::Trap(reason) => {
-                process_error(res.dispatch, initial_nonce, res.gas_amount.burned(), reason)
-            }
-            _ => process_success(res),
-        },
-        Err(e) => process_error(
-            dispatch,
-            initial_nonce,
-            e.gas_amount.burned(),
-            Some(e.reason),
-        ),
+    if let Some(program) = program {
+        let execution_settings = ExecutionSettings::new(block_info);
+        let initial_nonce = program.message_nonce();
+
+        let ret = match executor::execute_wasm::<E>(program, dispatch.clone(), execution_settings) {
+            Ok(res) => match res.kind {
+                DispatchResultKind::Trap(reason) => {
+                    process_error(res.dispatch, initial_nonce, res.gas_amount.burned(), reason)
+                }
+                _ => process_success(res),
+            },
+            Err(e) => process_error(
+                dispatch,
+                initial_nonce,
+                e.gas_amount.burned(),
+                Some(e.reason),
+            ),
+        };
+        return ret;
     }
+    process_skip(dispatch)
```

Comment:
Shouldn't it be under `else` block?

Reply:
Yeah

Reply:
you didn't consider return in the `if let`. I will define `else`, so it will be readable.